### PR TITLE
Rework Asset Store details page and navigation

### PR DESCRIPTION
**Pull request by ClementPasteau**
As part of reworking the Asset store, this PR aims to:

- improve the navigation between the 3 different screens:
  - Asset Home
  - Asset Search List
  - Asset Details
- improve the Asset Details page
  - No more a dialog
  - See the different animations
  - See a better feedback when adding an item
  - Be able to navigate between tags

testable with [Storybook](http://gdevelop-storybook.s3-website-us-east-1.amazonaws.com/improve-asset-details/latest/index.html?path=/story/assetstore-newobjectdialog--default)

Some screenshots of the current state.

Screenshot 2022-05-20 at 12 38 07
Screenshot 2022-05-20 at 12 36 47

### CODE REVIEW

**Comment by tristanbob** [comment]
Would it be possible to add left/right arrows to the sides of the animation drop-down? 

`<-  | Climb v |  ->`

Arrows = I want to quickly cycle through all animations
Dropdown = I want to know the names of all animations (and perhaps pick one)

**Comment by tristanbob** [comment]
"Enter your search"  -> "Search assets"

**Comment by 4ian** [comment]
> Would it be possible to add left/right arrows to the sides of the animation drop-down?
> 
> `<- | Climb v | ->`
> 
> Arrows = I want to quickly cycle through all animations Dropdown = I want to know the names of all animations (and perhaps pick one)

Sounds like a good idea. On the top, the preview will probably be displayed like it's done in the preview of a sprite (so you can see the animation and even zoom/dezoom). On the bottom, we can have next/previous buttons.

Also good suggestions for the placeholder/tab name 👍

**Comment by ClementPasteau** [comment]
Thanks for the feedback @tristanbob 👍

**Comment by Bouh** [comment]
For pixel art the preview is tiny, 
image


[Storybook](http://gdevelop-storybook.s3-website-us-east-1.amazonaws.com/improve-asset-details/latest/index.html?path=/story/assetstore-assetstore--default)

File: Core/GDCore/Extensions/Builtin/SpriteExtension/Direction.cpp [hunk 1]
```diff
@@ -29,6 +31,15 @@ const Sprite& Direction::GetSprite(std::size_t nb) const { return sprites[nb]; }
...
 Sprite& Direction::GetSprite(std::size_t nb) { return sprites[nb]; }
 
+const std::vector<gd::String>& Direction::GetSpriteNames() const {
+  static std::vector<gd::String> spriteNames;
+  spriteNames.clear();
+  for (std::size_t i = 0; i < sprites.size(); ++i) {
+    spriteNames.push_back(sprites[i].GetImageName());
+  }
+  return spriteNames;
+}
```
Comment (D8H) [inline Core/GDCore/Extensions/Builtin/SpriteExtension/Direction.cpp:41]: Can this be done with a `map()` JavaScript side?

**Comment by D8H** [comment]
LGTM

File: newIDE/app/src/AssetStore/AssetDetails.js [hunk 2]
```diff
@@ -139,79 +176,36 @@ export const AssetDetails = ({
...
 
+  // For sprite animations.
+  const assetAnimations = asset
+    ? asset.objectAssets[0].object.animations
+    : null;
+  const animation = assetAnimations
+    ? assetAnimations.find(({ name }) => name === selectedAnimationName)
+    : null;
+  const direction = animation ? animation.directions[0] : null;
+  const animationResources =
```
[the remark this answers is not in the document]
Comment (4ian) [reply newIDE/app/src/AssetStore/AssetDetails.js:188]: (Done)

File: newIDE/app/src/AssetStore/AssetDetails.js [hunk 3]
```diff
@@ -227,10 +218,184 @@ export const AssetDetails = ({
...
+                          value={animation.name}
+                          primaryText={
+                            makeFirstLetterUppercase(animation.name) ||
+                            t`Default` // Display default for animations with no name.
+                          }
+                        />
+                      ))}
+                    </SelectField>
+                    <div style={styles.arrowContainer}>
+                      <IconButton
```
Comment (4ian) [inline newIDE/app/src/AssetStore/AssetDetails.js:369]: I've added proper buttons here

File: newIDE/app/src/AssetStore/AssetDetails.js [hunk 4]
```diff
@@ -227,10 +221,182 @@ export const AssetDetails = ({
...
+                            ({ name }) => name === resourceName
+                          ).file;
+                        }}
+                        project={project}
+                        timeBetweenFrames={direction.timeBetweenFrames}
+                        isLooping // Always loop in the asset store.
+                        hideCheckeredBackground
+                        hideControls
+                        initialZoom={140 / Math.max(asset.width, asset.height)}
+                        fixedHeight={250}
```
[the remark this answers is not in the document]
Comment (4ian) [reply newIDE/app/src/AssetStore/AssetDetails.js]: (Done)

**Review by 4ian — COMMENTED** [review]
I've fixed a few UI stuff, be careful to always go for UI components (if you use a custom style => good chance there is something weird going on)

**Pull request merged**